# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MIRGraph.h
# Commit: 93842c3971f8
# Full Hash: 93842c3971f8e612537cba5d6d25b482c0d67a66
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-15 21:49:48
# Regressor Bug: 1601599
# File Overlap Count: 1
# Description:
#   Bug 1601599 part 3 - Simplify loop phi code in IonBuilder. r=tcampbell
#   
#   Before this bug:
#   
#   * We did not add phis for expression stack slots that aren't part of the loop.
# ==============================================================================

diff -r 4903b1b93526 -r 93842c3971f8 js/src/jit/MIRGraph.h
--- a/js/src/jit/MIRGraph.h	Sun Dec 15 11:39:00 2019 +0000
+++ b/js/src/jit/MIRGraph.h	Sun Dec 15 11:33:36 2019 +0000
@@ -44,8 +44,7 @@
   MOZ_MUST_USE bool init();
   void copySlots(MBasicBlock* from);
   MOZ_MUST_USE bool inherit(TempAllocator& alloc, size_t stackDepth,
-                            MBasicBlock* maybePred, uint32_t popped,
-                            unsigned stackPhiCount = 0);
+                            MBasicBlock* maybePred, uint32_t popped);
   MOZ_MUST_USE bool inheritResumePoint(MBasicBlock* pred);
   void assertUsesAreNotWithin(MUseIterator use, MUseIterator end);
 
@@ -123,8 +122,7 @@
   static MBasicBlock* NewPendingLoopHeader(MIRGraph& graph,
                                            const CompileInfo& info,
                                            MBasicBlock* pred,
-                                           BytecodeSite* site,
-                                           unsigned loopStateSlots);
+                                           BytecodeSite* site);
   static MBasicBlock* NewSplitEdge(MIRGraph& graph, MBasicBlock* pred,
                                    size_t predEdgeIdx, MBasicBlock* succ);
 
